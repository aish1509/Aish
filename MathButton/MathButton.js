import React from 'react';
import PropTypes from 'prop-types';

import {makeStyles} from '@material-ui/styles';

export const useStyles = makeStyles(theme => ({
	button: props => {
		const commonStyle = {
			borderRadius: '74px',
			width: '150px',
			height: '30px',
			fontFamily: theme.fonts.montserratBold,
			fontStyle: 'normal',
			fontWeight: 'bold',
			fontSize: '12px',
			cursor: props.disabled ? 'default' : 'pointer',
			letterSpacing: '0.015em',
			textTransform: 'capitalize'
		};

		const primaryStyle = {
			color: theme.colors.white,
			backgroundColor: props.disabled
				? theme.colors.blueGrey[60]
				: theme.colors.blue.dark,
			border: `none`,
			'&:hover': {
				backgroundColor: props.disabled
					? theme.colors.blueGrey[60]
					: theme.colors.blue.darkest
			}
		};

		const secondaryStyle = {
			color: props.disabled
				? theme.colors.blueGrey[60]
				: theme.colors.blue.dark,
			backgroundColor: theme.colors.white,
			border: `1.5px solid ${
				props.disabled
					? theme.colors.blueGrey[60]
					: theme.colors.blue.dark
			}`,
			'&:hover': {
				backgroundColor: props.disabled ? theme.colors.white : '#B3CFE2'
			}
		};

		return {
			...commonStyle,
			...(props.secondary ? secondaryStyle : primaryStyle),
			...props.buttonStyleModifier
		};
	}
}));

const MathButton = ({
	children,
	onClickHandler,
	disabled,
	buttonStyleModifier,
	secondary
}) => {
	const classes = useStyles({buttonStyleModifier, disabled, secondary});

	const handleButtonClick = e => {
		e.preventDefault();
		onClickHandler();
	};

	return (
		<button
			className={classes.button}
			disabled={disabled}
			onClick={handleButtonClick}
		>
			{children}
		</button>
	);
};

MathButton.defaultProps = {
	buttonStyleModifier: {},
	children: 'Go On',
	disabled: false,
	secondary: false
};

MathButton.propTypes = {
	children: PropTypes.node.isRequired,
	onClickHandler: PropTypes.func.isRequired,
	buttonStyleModifier: PropTypes.object,
	disabled: PropTypes.bool,
	secondary: PropTypes.bool
};

export default MathButton;
