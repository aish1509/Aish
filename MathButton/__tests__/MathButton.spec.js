import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';

import MathButton, {useStyles} from '../MathButton';

describe('<MathButton />', () => {
	let wrapper = null;

	let mockClickHandler = null;
	let mockEvent = null;

	beforeEach(() => {
		mockClickHandler = jest.fn();
		mockEvent = {preventDefault: jest.fn()};
		wrapper = shallow(
			<MathButton onClickHandler={mockClickHandler}>Go On</MathButton>
		);
	});

	describe('is not disabled', () => {
		beforeEach(() => {
			wrapper.setProps({disabled: false});
		});

		it('should render correctly', () => {
			expect(shallowToJson(wrapper)).toMatchSnapshot();
		});

		it('handleButtonClick', () => {
			const buttonComp = wrapper.find('button');
			buttonComp.simulate('click', mockEvent);
			expect(mockEvent.preventDefault).toHaveBeenCalled();
			expect(mockClickHandler).toHaveBeenCalled();
		});
	});

	describe('is disabled', () => {
		beforeEach(() => {
			wrapper.setProps({disabled: true});
		});

		it('should render correctly', () => {
			expect(shallowToJson(wrapper)).toMatchSnapshot();
		});
	});

	describe('useStyles', () => {
		describe('primary', () => {
			it('enabled', () => {
				expect(
					useStyles().button({disabled: false, secondary: false})
				).toMatchSnapshot();
			});

			it('disabled', () => {
				expect(
					useStyles().button({disabled: true, secondary: false})
				).toMatchSnapshot();
			});
		});

		describe('secondary', () => {
			it('enabled', () => {
				expect(
					useStyles().button({disabled: false, secondary: true})
				).toMatchSnapshot();
			});

			it('disabled', () => {
				expect(
					useStyles().button({disabled: true, secondary: true})
				).toMatchSnapshot();
			});
		});

		it('with buttonStyleModifier', () => {
			expect(
				useStyles().button({
					buttonStyleModifier: {width: '100px', height: '100px'},
					disabled: false,
					secondary: false
				})
			).toMatchSnapshot();
		});
	});
});
