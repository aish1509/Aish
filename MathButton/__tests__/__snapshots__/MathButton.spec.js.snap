// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MathButton /> is disabled should render correctly 1`] = `
<button
  className={[Function]}
  disabled={true}
  onClick={[Function]}
>
  Go On
</button>
`;

exports[`<MathButton /> is not disabled should render correctly 1`] = `
<button
  className={[Function]}
  disabled={false}
  onClick={[Function]}
>
  Go On
</button>
`;

exports[`<MathButton /> useStyles primary disabled 1`] = `
Object {
  "&:hover": Object {
    "backgroundColor": "#9BB1BD",
  },
  "backgroundColor": "#9BB1BD",
  "border": "none",
  "borderRadius": "74px",
  "color": "#FFFFFF",
  "cursor": "default",
  "fontFamily": undefined,
  "fontSize": "12px",
  "fontStyle": "normal",
  "fontWeight": "bold",
  "height": "30px",
  "letterSpacing": "0.015em",
  "textTransform": "capitalize",
  "width": "150px",
}
`;

exports[`<MathButton /> useStyles primary enabled 1`] = `
Object {
  "&:hover": Object {
    "backgroundColor": "#003F6A",
  },
  "backgroundColor": "#005E9D",
  "border": "none",
  "borderRadius": "74px",
  "color": "#FFFFFF",
  "cursor": "pointer",
  "fontFamily": undefined,
  "fontSize": "12px",
  "fontStyle": "normal",
  "fontWeight": "bold",
  "height": "30px",
  "letterSpacing": "0.015em",
  "textTransform": "capitalize",
  "width": "150px",
}
`;

exports[`<MathButton /> useStyles secondary disabled 1`] = `
Object {
  "&:hover": Object {
    "backgroundColor": "#FFFFFF",
  },
  "backgroundColor": "#FFFFFF",
  "border": "1.5px solid #9BB1BD",
  "borderRadius": "74px",
  "color": "#9BB1BD",
  "cursor": "default",
  "fontFamily": undefined,
  "fontSize": "12px",
  "fontStyle": "normal",
  "fontWeight": "bold",
  "height": "30px",
  "letterSpacing": "0.015em",
  "textTransform": "capitalize",
  "width": "150px",
}
`;

exports[`<MathButton /> useStyles secondary enabled 1`] = `
Object {
  "&:hover": Object {
    "backgroundColor": "#B3CFE2",
  },
  "backgroundColor": "#FFFFFF",
  "border": "1.5px solid #005E9D",
  "borderRadius": "74px",
  "color": "#005E9D",
  "cursor": "pointer",
  "fontFamily": undefined,
  "fontSize": "12px",
  "fontStyle": "normal",
  "fontWeight": "bold",
  "height": "30px",
  "letterSpacing": "0.015em",
  "textTransform": "capitalize",
  "width": "150px",
}
`;

exports[`<MathButton /> useStyles with buttonStyleModifier 1`] = `
Object {
  "&:hover": Object {
    "backgroundColor": "#003F6A",
  },
  "backgroundColor": "#005E9D",
  "border": "none",
  "borderRadius": "74px",
  "color": "#FFFFFF",
  "cursor": "pointer",
  "fontFamily": undefined,
  "fontSize": "12px",
  "fontStyle": "normal",
  "fontWeight": "bold",
  "height": "100px",
  "letterSpacing": "0.015em",
  "textTransform": "capitalize",
  "width": "100px",
}
`;
